src/amp-story-player/amp-story-player-impl.js

```diff
@@ -221,14 +221,18 @@ export class AmpStoryPlayer {
   initializeShadowRoot_() {
     this.rootEl_ = this.doc_.createElement('main');
 
-    // Create shadow root
-    const shadowRoot = this.element_.attachShadow({mode: 'open'});
+    // TODO(): Update unit tests to work without shadow root,
+    // then update condition to getMode.test and update visual tests.
+    const containerToUse =
+      typeof __AMP_VISUAL_TEST !== 'undefined'
```

Comment:
I believe this feature already exists as `__AMP_TEST`, but either @choumx or @newmuis will know better than me :)

Reply:
Yep, it's a bug on the stories side; the player unit tests currently (incorrectly) *require* shadow DOM to be enabled, so we need to disable shadow DOM for visual tests but not unit tests.

We should file a GitHub issue to do that work and include the issue number here (e.g. `TODO(#12345): ...`)

Reply:
I think using this helper instead should work: https://github.com/ampproject/amphtml/blob/5059ece5035547b50530714fe443ba41c9223b69/extensions/amp-story/1.0/utils.js#L95-L112